src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.GCDesc.cs

```diff
@@ -0,0 +1,200 @@
+                            if (defType.ContainsPointers)
+                            {
+                                int numSeries = GCPointerMap.FromInstanceLayout(defType).NumSeries;
+                                Debug.Assert(numSeries > 0);
+                                return (numSeries + 2) * _type.Context.Target.PointerSize;
+                            }
+                            return 0;
+                        }
+                    }
+                }
+                else
+                {
+                    var defType = (DefType)_type;
+                    if (defType.ContainsPointers)
+                    {
+                        int numSeries = GCPointerMap.FromInstanceLayout(defType).NumSeries;
+                        Debug.Assert(numSeries > 0);
+                        return (numSeries * 2 + 1) * _type.Context.Target.PointerSize;
+                    }
+                    return 0;
+                }
+            }
+        }
+
+        private void OutputGCDesc(ref ObjectDataBuilder builder)
+        {
+            if (!_constructed || _type.IsGenericDefinition)
+            {
+                Debug.Assert(GCDescSize == 0);
+                return;
+            }
+
+            int initialBuilderPosition = builder.CountBytes;
+
+            if (_type.IsArray)
+            {
+                TypeDesc elementType = ((ArrayType)_type).ElementType;
+                if (elementType.IsObjRef)
+                {
+                    OutputStandardGCDesc(ref builder,
+                        new GCPointerMap(new[] { 1 << 2 }, 3),
+                        4 * _type.Context.Target.PointerSize, 0);
+                }
+                else
+                {
+                    if (elementType.IsPointer)
+                    {
+                        // Nothing to output here. Unmanaged pointers don't point to the GC heap.
+                    }
+                    else if (elementType.IsByRef)
+                    {
+                        throw new BadImageFormatException();
+                    }
+                    else
+                    {
+                        var elementDefType = (DefType)elementType;
+                        if (elementDefType.ContainsPointers)
+                        {
+                            OutputArrayGCDesc(ref builder, GCPointerMap.FromInstanceLayout(elementDefType));
+                        }
+                    }
+                }
+            }
+            else
+            {
+                var defType = (DefType)_type;
+                if (defType.ContainsPointers)
+                {
+                    // Computing the layout for the boxed version if this is a value type.
+                    int offs = defType.IsValueType ? _type.Context.Target.PointerSize : 0;
+
+                    // Include syncblock
+                    int allocationSize = defType.InstanceByteCount + offs + _type.Context.Target.PointerSize;
```

Comment:
This should be same as the math `OutputBaseSize` - it would be nice to make it look same.